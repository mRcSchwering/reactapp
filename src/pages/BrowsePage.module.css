.SearchPanel {
  height: 80px;
  background-color: gainsboro;
  display: flex;
  justify-content: center;
  align-items: center;
}
