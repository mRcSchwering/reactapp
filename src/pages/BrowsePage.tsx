import React from "react";
import styles from "./BrowsePage.module.css";
import { useLocation } from "react-router-dom";

function useQueryStringParams() {
  return new URLSearchParams(useLocation().search);
}

function SearchPanel() {
  return (
    <div className={styles.SearchPanel}>
      <div>The search panel</div>
    </div>
  );
}

export default function BrowsePage(): JSX.Element {
  const queryStringParams = useQueryStringParams();
  const search = queryStringParams.get("search");

  return (
    <div>
      <SearchPanel />
      <h2>Browse Page</h2>
      <p>Searched: {search}</p>
    </div>
  );
}
