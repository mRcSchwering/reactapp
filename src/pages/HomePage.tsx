import React from "react";
import { Form, Button } from "react-bootstrap";
import { useFetchHackerNewsData } from "../hooks/useFetchData";
import { ErrorModalContext } from "../components/ErrorModalContext";

export default function HomePage() {
  const { showErrorModal } = React.useContext(ErrorModalContext);
  const [query, setQuery] = React.useState(8863);
  const resp = useFetchHackerNewsData(query);

  function handleInputChange(value: string) {
    setQuery(parseInt(value, 10));
  }

  function handleButtonClick() {
    showErrorModal("button was clicked");
  }

  if (resp.loading) {
    return <div>Loading....</div>;
  }

  if (resp.error) {
    showErrorModal("loading error");
  }

  return (
    <div>
      <h2>Home Page</h2>
      <h4>Response</h4>
      <div style={{ width: "200px", margin: "auto" }}>
        <Form.Control
          type="number"
          value={query}
          onChange={(e) => handleInputChange(e.currentTarget.value)}
        />
      </div>
      <div>{JSON.stringify(resp.data)}</div>
      <Button onClick={handleButtonClick}>show error modal</Button>
    </div>
  );
}
