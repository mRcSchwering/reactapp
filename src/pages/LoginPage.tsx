import React from "react";

export default function LoginPage(): JSX.Element {
  return (
    <div>
      <h2>Login Page</h2>
    </div>
  );
}
