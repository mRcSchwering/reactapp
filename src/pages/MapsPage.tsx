import React from "react";
import { GoogleMapEmbed } from "../components/maps";

export default function MapsPage() {
  return (
    <div>
      <h2>Maps Page</h2>
      <h4>Google Maps Embed</h4>
      <GoogleMapEmbed />
    </div>
  );
}
