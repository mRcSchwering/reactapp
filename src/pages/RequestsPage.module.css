.inputContainer {
  width: 200px;
  margin: auto;
}

.resultsContainer {
  max-width: 1000px;
  min-width: 600px;
  padding: 10px;
  margin: auto;
  text-align: left;
}
