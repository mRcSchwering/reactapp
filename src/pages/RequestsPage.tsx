import React from "react";
import { Form, Button } from "react-bootstrap";
import { useFetchHackerNewsData } from "../hooks/useFetchData";
import { gql } from "apollo-boost";
import { useQuery } from "@apollo/react-hooks";
import styles from "./RequestsPage.module.css";

type RamCharacterDataType = {
  character: {
    name: string;
    species: string;
    type: string;
    gender: string;
    origin: {
      name: string;
      dimension: string;
    };
    status: string;
  };
};

type RamCharacterVarsType = {
  id: number;
};

const RAM_CHARACTER_QUERY = gql`
  query RamChars($id: ID!) {
    character(id: $id) {
      name
      species
      type
      gender
      origin {
        name
        dimension
      }
      status
    }
  }
`;

function getResultElement(res: any): JSX.Element {
  if (res.loading) return <>loading...</>;
  if (res.error) return <>error: {JSON.stringify(res.error, null, 2)} </>;
  return <>{JSON.stringify(res.data, null, 2)}</>;
}

export default function RequestsPage() {
  const [hnId, setHnId] = React.useState(8863);
  const hn = useFetchHackerNewsData(hnId);

  const [ramId, setRamId] = React.useState(1);
  const [queryRamId, setQueryRamId] = React.useState(ramId);
  const ram = useQuery<RamCharacterDataType, RamCharacterVarsType>(
    RAM_CHARACTER_QUERY,
    { variables: { id: queryRamId } }
  );

  function handleButtonClick() {
    setQueryRamId(ramId);
  }

  const hnResults = getResultElement(hn);
  const ramResults = getResultElement(ram);

  return (
    <div>
      <h2>Requests Page</h2>
      <h4>Hacker News REST API</h4>
      <p>Load on input change. Uses custom hook for fetching.</p>
      <div
        style={{
          height: "200px",
          width: "1000px",
          margin: "auto",
          padding: "10px",
        }}
      >
        <div className={styles.inputContainer}>
          <Form.Control
            type="number"
            value={hnId}
            onChange={(e) => setHnId(parseInt(e.currentTarget.value, 10))}
          />
        </div>
        <div className={styles.resultsContainer}>{hnResults}</div>
      </div>
      <div>
        <h4>Rick and Morty GraphQL API</h4>
        <p>
          Load on button click. The Apollo Provider (graphQL client) is mounted
          in App.tsx
        </p>
        <div className={styles.inputContainer}>
          <Form.Control
            type="number"
            value={ramId}
            onChange={(e) => setRamId(parseInt(e.currentTarget.value, 10))}
          />
          <Button onClick={handleButtonClick}>Send</Button>
        </div>
        <div className={styles.resultsContainer}>
          <pre>{ramResults}</pre>
        </div>
      </div>
    </div>
  );
}
