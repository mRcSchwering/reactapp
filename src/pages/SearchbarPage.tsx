import React from "react";
import { useDummyFetch } from "../hooks/useDummyFetch";

export function getResultElement(res: { loading; error?; data }): JSX.Element {
  if (res.loading) return <>loading...</>;
  if (res.error) return <>error: {JSON.stringify(res.error, null, 2)} </>;
  return <>{JSON.stringify(res.data, null, 2)}</>;
}

/**
 * TODO: cant avoid first render, see use DummyFetch
 */
export default function SearchbarPage() {
  console.log("search page");
  const [value, setValue] = React.useState("");
  const [search, setSearch] = React.useState("");
  const apiState = useDummyFetch(search);

  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
    setValue(e.target.value);
  }

  React.useEffect(() => {
    if (value.length >= 3) {
      setSearch(value);
    }
  }, [value]);

  return (
    <div>
      <h2>Searchbar Page</h2>
      <h4>Searching random APIs</h4>
      <form>
        <input
          type="text"
          placeholder="Enter something..."
          onChange={handleInputChange}
          value={value}
        />
      </form>
      <h4>Results</h4>
      <div>
        <pre>{getResultElement(apiState)}</pre>
      </div>
    </div>
  );
}
