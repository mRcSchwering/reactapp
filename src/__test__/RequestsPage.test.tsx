import { getResultElement } from "../pages/RequestsPage";

it("shows loading fragment", () => {
  const data = { loading: true, error: null, data: null };
  const res = getResultElement(data);
  expect(res).toMatchSnapshot();
});

it("shows error fragment", () => {
  const data = { loading: false, error: "test error", data: null };
  const res = getResultElement(data);
  expect(res).toMatchSnapshot();
});

it("shows data fragment", () => {
  const data = { loading: false, error: null, data: { a: [1, 2] } };
  const res = getResultElement(data);
  expect(res).toMatchSnapshot();
});
