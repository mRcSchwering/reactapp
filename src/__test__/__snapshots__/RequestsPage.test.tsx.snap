// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows data fragment 1`] = `
<React.Fragment>
  {
  "a": [
    1,
    2
  ]
}
</React.Fragment>
`;

exports[`shows error fragment 1`] = `
<React.Fragment>
  error: 
  "test error"
   
</React.Fragment>
`;

exports[`shows loading fragment 1`] = `
<React.Fragment>
  loading...
</React.Fragment>
`;
